### ISSUE TITLE
Validation accuracy is incorrect when using fully convolutional models

### ISSUE THREAD

**Issue by nshaud** [nshaud/DeepHyperX#24]

**Comment by mengxue-rs** [comment]
Hi, which one is you refering to? I would like to try it.

**Comment by nshaud** [comment]
@snowzm you can try `lee` model which is fully convolutional IIRC. Validation accuracy is grossly incorrect in this case.

**Comment by mengxue-rs** [comment]
@nshaud I think this issue may relate to your choice of the normalization method. More details see [Experiment Reports.pdf](https://github.com/nshaud/DeepHyperX/files/5601022/Experiment.Reports.pdf)

**Comment by mengxue-rs** [comment]
@nshaud there may be two reasons about this issue. 
1) in the **line 1225** of the **val** function of the **models.py**, `if out.item() in ignored_labels:` may be corrected as `if pred.item() in ignored_labels:`. This results in the incorrect calculation of the validation accuracy; 
2) wrongly used normalization makes the training loss being still high, you could try the **SNB normalization** (see my recent pulled request).
There are three experimental pictures about above statements (10% samples per class on the Indian Pines data set). When using original settings, I got the first row images. Then I got the second row images if I did 1). Finally, I got the last row images if I did 1) and 2).

![1](https://user-images.githubusercontent.com/22807370/100595830-6921b780-3336-11eb-9df9-644630c20542.png)
![2](https://user-images.githubusercontent.com/22807370/100595836-6b841180-3336-11eb-9fa9-a60dc3d7cc8f.png)
![3](https://user-images.githubusercontent.com/22807370/100595851-6e7f0200-3336-11eb-9940-9f0900205dbf.png)
